llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -173,6 +173,12 @@ Value *llvm::findScalarElement(Value *V, unsigned EltNo) {
   if (Constant *C = dyn_cast<Constant>(V))
     return C->getAggregateElement(EltNo);
 
+  if (auto *F = dyn_cast<FreezeInst>(V)) {
+    if (isGuaranteedNotToBeUndefOrPoison(F)) {
```

Comment:
Won't this always return true? You're passing in the freeze itself. This should be checking F->getOperand(0)?